identity-server/src/IdentityServer/Validation/Default/PrivateKeyJwtSecretValidator.cs

```diff
@@ -61,7 +63,15 @@ public async Task<SecretValidationResult> ValidateAsync(IEnumerable<Secret> secr
         var fail = new SecretValidationResult { Success = false };
         var success = new SecretValidationResult { Success = true };
 
-        if (parsedSecret.Type != ParsedSecretTypes.JwtBearer)
+        // Decide whether to enforce strict audience validation or not.
+        var enforceStrictAud = _options.Preview.StrictClientAssertionAudienceValidation;
+
+        if (enforceStrictAud && parsedSecret.Type != ParsedSecretTypes.ClientAuthenticationJwt)
```

Comment:
We can't use the `parsedSecret.Type` field in this way. That type is a property that is saying what kind of authentication the client has used. If it happens to be a jwt, then we still need to validate the jwt to get at its typ header.